Taskfile.yml

```diff
@@ -324,42 +327,55 @@ tasks:
       # -race fails at the moment in gopter - possibly due to our shared generator variable?
       - go test -short -tags=noexit -timeout 10m -covermode atomic -coverprofile=controller-coverage.out -json -coverpkg="./..." -run '{{default ".*" .TEST_FILTER}}' ./... > '{{.TEST_OUT}}/controller-unit-tests.json'
 
-  controller:test-upgrade-pre:
-    desc: Test upgrading {{.CONTROLLER_APP}} and helm chart for new release
+  controller:test-upgrade-setup:
+    desc: Setup for upgrade test
     dir: "{{.CONTROLLER_ROOT}}"
     cmds:
-      - task: controller:kind-create
+      - task: controller:kind-create-wi
+      - task: controller:create-mi-for-workload-identity
       - task: controller:install-cert-manager
       - task: controller:docker-push-local
       - task: controller:gen-helm-manifest
```

Comment:
Do these have to be serial like this? We should be preferring `deps` over calling tasks like this, where possible.

Reply:
The bits I added/swapped (king-create-wi and create-mi-for-workload-identity) do need to be run in serial, yes, as the first produces output consumed by the second.

install-cert-manager also must run after the first 2.

We could in theory move `docker-push-local` and `gen-helm-manifest` to deps, but my focus here was on changing auth not necessarily improving build parallelism and the parts that have been changed do need to run in sequence. Given the priority on this auth stuff I think it makes sense to focus on that for this PR as I am not 100% sure that moving other things around to deps/etc will trivially work.